contracts/SnappBase.sol

```diff
@@ -117,6 +116,7 @@ contract SnappBase is Ownable {
 
         stateRoots.push(_newStateRoot);        
         depositHashes[slot].applied = true;
-        emit StateTransition(TransitionType.Deposit, _currStateRoot, _newStateRoot, slot);
+        emit StateTransition(TransitionType.Deposit, this.stateIndex(), _newStateRoot, slot);
```

Comment:
the "this" is not needed, right?

Reply:
State index is a public view function returnin the length of state array minus one.

Reply:
That is excellent, I have removed all uses of this.publicViewFunction